addons/website_sale_delivery_mondialrelay/controllers/controllers.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+from odoo import http, _
+from odoo.addons.website_sale.controllers.main import WebsiteSale, PaymentPortal
+from odoo.addons.website_sale_delivery.controllers.main import WebsiteSaleDelivery
+
+from odoo.exceptions import AccessDenied, ValidationError, UserError
+from odoo.http import request
+
+class MondialRelay(http.Controller):
+
+    @http.route(['/website_sale_delivery_mondialrelay/update_shipping'], type='json', auth="public", website=True, multilang=False)
+    def mondial_relay_update_shipping(self, **data):
+        order = request.website.sale_get_order()
+
+        if order.partner_id == request.website.user_id.sudo().partner_id:
+            raise AccessDenied('Customer of the order cannot be the public user at this step.')
```

Comment:
It means that it is not possible to buy and delivery with MR without account ?
I think this check can be removed, no ?

Reply:
no, it just means that at this step, you should have at least a billing address on your order.